### PR TITLE
Improve capacity plugin: Only compare requested dimension resources when reclaim

### PR DESCRIPTION
**Pull request by Monokaix**
1. Part of https://github.com/volcano-sh/volcano/issues/3537, improve AddPreemptive func
2. Add `reclaim` action of capacity plugin user guide.

### CODE REVIEW

**Comment by Monokaix** [comment]
cc @lowang-bh

File: pkg/scheduler/api/types.go [hunk 1]
```diff
@@ -119,6 +119,9 @@ type CompareFn func(interface{}, interface{}) int
 // ValidateFn is the func declaration used to check object's status.
 type ValidateFn func(interface{}) bool
 
+// ValidateWithCandidateFn behaves like ValidateFn but take the candidate task into consideration.
+type ValidateWithCandidateFn func(interface{}, interface{}) bool
```
[the remark this answers is not in the document]
Comment (Monokaix) [reply pkg/scheduler/api/types.go:123]: `Allocatable` also use candidate, I think we can keep consistent with it.

**Comment by Monokaix** [comment]
cc @hwdef

**Review by hwdef — COMMENTED** [review]
/approve

**Pull request merged**